providers.d/apt-mirror/provider_wrapper.sh

```diff
@@ -1,13 +1,37 @@
 bashelliteProviderWrapper() {
 
-  if [[ ! ${_r_dryrun} ]]; then
-    apt-mirror ${_r_metadata_tld}/repos.conf.d/${_n_repo_name}/provider.conf;
+  local basepath_parameter="$(grep -oP "(?<=(^set base_path[[:space:]]*))[[:alnum:]/]*" ${_r_metadata_tld}/repos.conf.d/${_n_repo_name}/provider.conf)"
+  if [[  "${basepath_parameter}" != "${_r_mirror_tld}/${_n_repo_name}" ]]; then
+    utilMsg WARN "$(utilTime)" "The \"base_path\" parameter (${basepath_parameter}) in provider.conf does not match mirror location (${_r_mirror_tld}/${_n_repo_name})..."
   fi
 
-  if [[ "${?}" != "0" ]]; then
-    utilMsg WARN "$(utilTime)" "apt-mirror either failed or completed with errors for repo (${_n_repo_name})."
+  local config_file="${_r_metadata_tld}/repos.conf.d/${_n_repo_name}/provider.conf"
+  local tmp_config_file="${HOME}/.bashellite/tmp_${_n_repo_name}_provider.conf"
+
+  cat /dev/null > ${tmp_config_file}
+
+  IFS=$'\n'
+  for line in $(cat ${config_file}); do
```

Comment:
No need to call cat. Use input redirection:

```
while IFS=$'\n' read line; do 
     ### do your commands
done < ${config_file}
```

For example:

```
root@chrx:~# for n in {1..9}; do echo "line$n,column1 line$n,column2" >> test.txt; done && cat test.txt
line1,column1 line1,column2
line2,column1 line2,column2
line3,column1 line3,column2
line4,column1 line4,column2
line5,column1 line5,column2
line6,column1 line6,column2
line7,column1 line7,column2
line8,column1 line8,column2
line9,column1 line9,column2
root@chrx:~# input_file="test.txt"; while IFS=$'\n' read line; do echo $line; done < $input_file
line1,column1 line1,column2
line2,column1 line2,column2
line3,column1 line3,column2
line4,column1 line4,column2
line5,column1 line5,column2
line6,column1 line6,column2
line7,column1 line7,column2
line8,column1 line8,column2
line9,column1 line9,column2
```

Reply:
Will go ahead and make this change.  Both here and in the bandersnatch provider PR.